internal/api/v1/application/stage.go

```diff
@@ -134,6 +134,48 @@ func (hc Controller) Stage(c *gin.Context) apierror.APIErrors {
 		return apierror.NewBadRequest("pipelinerun for image ID still running")
 	}
 
+	// Validate incoming blob id before attempting to stage
+
+	s3ConnectionDetails, err := s3manager.GetConnectionDetails(ctx, cluster, deployments.TektonStagingNamespace, deployments.S3ConnectionDetailsSecret)
+	if err != nil {
+		return apierror.InternalError(err, "failed to fetch the S3 connection details")
+	}
+
+	manager, err := s3manager.New(s3ConnectionDetails)
```

Comment:
All the additions below this point can be extracted to a function that does what the comment above says. E.g.

```
validateBlob(req.BlobUID, req.App, s3ConnectionDetails)
```

Reply:
Will do.

Reply:
commit 189a36981220ee32960d620612dadaf51204995d